client/changelist/files_changelist.go

```diff
@@ -0,0 +1,103 @@
+package changelist
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"time"
+
+	"code.google.com/p/go-uuid/uuid"
+)
+
+type fileChangelist struct {
+	dir string
+}
+
+func NewFileChangelist(dir string) (*fileChangelist, error) {
+	err := os.MkdirAll(dir, 0600)
+	if err != nil {
+		return nil, err
+	}
+	return &fileChangelist{dir: dir}, nil
+}
+
+func (cl fileChangelist) List() []Change {
+	changes := make([]Change, 0)
+	dir, err := os.Open(cl.dir)
+	if err != nil {
+		return changes
+	}
+	fileInfos, err := dir.Readdir(0)
+	if err != nil {
+		return changes
+	}
+	sort.Sort(fileChanges(fileInfos))
+	for _, f := range fileInfos {
+		if f.IsDir() {
+			continue
+		}
+		raw, err := ioutil.ReadFile(path.Join(cl.dir, f.Name()))
+		if err != nil {
+			// TODO(david): How should we handle this?
```

Comment:
I think we need to fail hard if theres a corrupted/unreadable file in this dir. Otherwise we risk losing data.